model/textparse/openmetricsparse_test.go

```diff
@@ -64,7 +64,48 @@ _metric_starting_with_underscore 1
 testmetric{_label_starting_with_underscore="foo"} 1
 testmetric{label="\"bar\""} 1
 # TYPE foo counter
-foo_total 17.0 1520879607.789 # {id="counter-test"} 5`
+foo_total 17.0 1520879607.789 # {id="counter-test"} 5
+foo_created 1000
+foo_total{a="b"} 17.0 1520879607.789 # {id="counter-test"} 5
+foo_created{a="b"} 1000
+# TYPE bar summary
+bar_count 17.0
+bar_sum 324789.3
+bar{quantile="0.95"} 123.7
+bar{quantile="0.99"} 150.0
+bar_created 1520430000
+# TYPE baz histogram
+baz_bucket{le="0.0"} 0
+baz_bucket{le="+Inf"} 17
+baz_count 17
+baz_sum 324789.3
+baz_created 1520430000
+# TYPE something histogram
+something_count 17
+something_sum 324789.3
+something_created 1520430000
+something_bucket{le="0.0"} 0
+something_bucket{le="+Inf"} 17
+# TYPE thing histogram
+thing_created 1520430000
+thing_count 17
+thing_sum 324789.3
+thing_bucket{le="0.0"} 0
+thing_bucket{le="+Inf"} 17
+# TYPE yum summary
+yum_count 17.0
+yum_sum 324789.3
+yum_created 1520430000
+yum{quantile="0.95"} 123.7
+yum{quantile="0.99"} 150.0
+# TYPE foobar summary
+foobar_created 1520430000
+foobar_count 17.0
+foobar_sum 324789.3
+foobar{quantile="0.95"} 123.7
+foobar{quantile="0.99"} 150.0
+# TYPE fizz_created gauge
+fizz_created 17.0`
```

Comment:
All tests here fail, right? The current implementation doesn't work if _created is not the last line and we are aware that we skip _created lines even if they are gauges.

If I remember correctly in our discussion tomorrow with @bwplotka and @TheSpiritXIII, we're ok with proceeding this way, but let's add clear tests showing that this is the expected behavior for now. I'm assuming this is why you added those tests, right?

It's good the they really show tests are failing! Goal completed! But we're blocking CI now 😅. Would it make sense to create a separate test, like `TestCTParsingFailures` that will look very much like this test, but at the end we assert failures instead of success? This way we prove the failure modes, but we don't block CI

Reply:
Done! Moved these tests to a new function which checks if an error is expected, then returned ct != expected ct